src/main/java/slimeknights/tconstruct/tables/network/UpdateStationScreenPacket.java

```diff
@@ -0,0 +1,71 @@
+package slimeknights.tconstruct.tables.network;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraftforge.fml.network.NetworkEvent;
+import slimeknights.mantle.network.packet.IThreadsafePacket;
+import slimeknights.tconstruct.tables.client.inventory.BaseStationScreen;
+
+public class UpdateStationScreenPacket implements IThreadsafePacket {
+
+  private final PacketType type;
+  private final ITextComponent message;
+
+  /**
+   *
+   * @param type the type of sync the packet is, can be NO_TYPE, ERROR OR WARNING
+   * @param message the message to display if a warning or error occurred
+   */
+  public UpdateStationScreenPacket(PacketType type, ITextComponent message) {
+    this.type = type;
+    this.message = message;
+  }
+
+  public UpdateStationScreenPacket(PacketBuffer buffer) {
+    this.type = PacketType.valueOf(buffer.readString());
+    this.message = buffer.readTextComponent();
+  }
+
+  @Override
+  public void encode(PacketBuffer packetBuffer) {
+    packetBuffer.writeString(this.type.toString());
+    packetBuffer.writeTextComponent(this.message);
```

Comment:
I wonder if its worth saving some packet space by not syncing the message if its `StringTextComponent.EMPTY`. You can do that with the following:

```java
if (message == StringTextComponent.EMPTY) {
  buffer.writeBoolean(false);
} else {
  buffer.writeBoolean(true);
  buffer.writeTextComponent(this.message);
}
```

Then read with:
```java
if (buffer.readBoolean()) {
  this.message = buffer.readTextComponent();
} else {
  this.message = StringTextComponent.EMPTY;
}
```